lstchain/image/cleaning.py

```diff
@@ -36,3 +50,211 @@ def apply_dynamic_cleaning(image, signal_pixels, threshold, fraction):
     mask_dynamic_cleaning = (image >= dynamic_threshold) & signal_pixels
 
     return mask_dynamic_cleaning
+
+
+def find_tailcuts(input_dir, run_number):
+    """
+    This functions uses DL1 files to determine tailcuts which are adequate
+    for the bulk of the pixels in a given run. It does so simply based on the
+    median (for the whole camera) of the median pixel charge for pedestal
+    events.
+    For reasons of stability & simplicity of analysis, we cannot decide the
+    cleaning levels on a subrun-by-subrun basis. We select values which are
+    valid for the whole run.
+    The script also returns the suggested NSB adjustment needed in the
+    "dark-sky" MC to match the data.
```

Comment:
Is the returned rate to be applied at the waveform level ?

Reply:
Yes, I will clarify it.